src/python/director/motionplanningpanel.py

```diff
@@ -0,0 +1,391 @@
+        if terrain:
+
+            pos = np.array(frameObj.transform.GetPosition())
+
+            polyData = filterUtils.removeNonFinitePoints(terrain.polyData)
+            if polyData.GetNumberOfPoints():
+                polyData = segmentation.labelDistanceToLine(polyData, pos, pos+[0,0,1])
+                polyData = segmentation.thresholdPoints(polyData, 'distance_to_line', [0.0, 0.1])
+                if polyData.GetNumberOfPoints():
+                    pos[2] = np.nanmax(vnp.getNumpyFromVtk(polyData, 'Points')[:,2])
+                    frameObj.transform.Translate(pos - np.array(frameObj.transform.GetPosition()))
+
+            d = DebugData()
+            d.addSphere((0,0,0), radius=0.03)
+            handle = vis.showPolyData(d.getPolyData(), 'walking goal terrain handle', parent=frameObj, visible=True, color=[1,1,0])
+            handle.actor.SetUserTransform(frameObj.transform)
+            self.placer = PlacerWidget(app.getCurrentRenderView(), handle, terrain)
+
+            def onFramePropertyModified(propertySet, propertyName):
+                if propertyName == 'Edit':
+                    if propertySet.getProperty(propertyName):
+                        self.placer.start()
+                    else:
+                        self.placer.stop()
+
+            frameObj.properties.connectPropertyChanged(onFramePropertyModified)
+            onFramePropertyModified(frameObj, 'Edit')
+
+        frameObj.connectFrameModified(self.onWalkingGoalModified)
+        self.onWalkingGoalModified(frameObj)
+        
+    def onWalkingGoalModified(self, frame):
+        om.removeFromObjectModel(om.findObjectByName('footstep widget'))
+        request = self.footDriver.constructFootstepPlanRequest(self.robotStateJointController.q, frame.transform)
+        self.footDriver.sendFootstepPlanRequest(request)
+        
+    def onMotionPlan(self):
+        startPoseName = 'reach_start'
+        startPose = np.array(self.robotStateJointController.q)
+        self.ikPlanner.addPose(startPose, startPoseName)
+        plan = self.constraintSet.runIkTraj()
+        self.showPlan(plan)
+        
+    def hideTeleopModel(self):
+        self.teleopRobotModel.setProperty('Visible', False)
+        self.robotStateModel.setProperty('Visible', True)
+        self.robotStateModel.setProperty('Alpha', 1.0)
+        
+    def showPlan(self, plan):
+        self.hideTeleopModel()
+        self.showPlanFunction(plan)
+def _getAction():
+    return app.getToolBarActions()['ActionMotionPlanningPanel']
+
+def init(robotStateModel, robotStateJointController, teleopRobotModel, teleopJointController, debrisPlanner, manipPlanner, affordanceManager, showPlanFunction, hidePlanFunction, footDriver):
+
+    global panel
+    global dock
+
+    panel = MotionPlanningPanel(robotStateModel, robotStateJointController, teleopRobotModel, teleopJointController, debrisPlanner, manipPlanner, affordanceManager, showPlanFunction, hidePlanFunction, footDriver)
+    panel.deactivate()
```

Comment:
is this call to deactivate needed? it would be nice to make sure all the initialization if done in the MotionPlanningPanel constructor